### PR TITLE
executor: log inconsistent handles when inconsistent-check fail in `IndexLookupReader`

### PR DESCRIPTION
**Pull request by qw4990**
### What problem does this PR solve? 
Log missing handles and total handles returned by `IndexScan` when inconsistent-check fail.

### What is changed and how it works?
Log when inconsistent-check fail.

### CODE REVIEW

File: executor/distsql.go [hunk 1]
```diff
@@ -839,10 +839,16 @@ func (w *tableWorker) executeTask(ctx context.Context, task *lookupTableTask) er
...
+				handle := row.GetInt64(w.handleIdx)
+				obtainedHandlesMap[handle] = struct{}{}
+			}
+
 			// table scan in double read can never has conditions according to convertToIndexScan.
 			// if this table scan has no condition, the number of rows it returns must equal to the length of handles.
-			return errors.Errorf("inconsistent index %s handle count %d isn't equal to value count %d",
-				w.idxLookup.index.Name.O, handleCnt, len(task.rows))
+			return errors.Errorf("inconsistent index %s handle count %d isn't equal to value count %d, missing handles %v, total handles %v",
+				w.idxLookup.index.Name.O, handleCnt, len(task.rows), GetLackHandles(task.handles, obtainedHandlesMap), GetLackHandles(task.handles, obtainedHandlesMap), task.handles)
```
Comment (jackysp) [inline executor/distsql.go]:
```suggestion
				w.idxLookup.index.Name.O, handleCnt, len(task.rows), GetLackHandles(task.handles, obtainedHandlesMap), task.handles)
```
duplicated?

Comment (qw4990) [reply executor/distsql.go]: Yes... It fixed now.

File: executor/distsql.go [hunk 2]
```diff
@@ -839,10 +839,16 @@ func (w *tableWorker) executeTask(ctx context.Context, task *lookupTableTask) er
...
+				handle := row.GetInt64(w.handleIdx)
+				obtainedHandlesMap[handle] = struct{}{}
+			}
+
 			// table scan in double read can never has conditions according to convertToIndexScan.
 			// if this table scan has no condition, the number of rows it returns must equal to the length of handles.
-			return errors.Errorf("inconsistent index %s handle count %d isn't equal to value count %d",
-				w.idxLookup.index.Name.O, handleCnt, len(task.rows))
+			return errors.Errorf("inconsistent index %s handle count %d isn't equal to value count %d, missing handles %v, total handles %v",
+				w.idxLookup.index.Name.O, handleCnt, len(task.rows), GetLackHandles(task.handles, obtainedHandlesMap), task.handles)
```
Comment (jackysp) [inline executor/distsql.go]: If the map is large, will the error message become too long? Maybe it is better to print a log for it.

Comment (qw4990) [reply executor/distsql.go]: updated.

File: executor/distsql.go [hunk 3]
```diff
@@ -839,6 +839,17 @@ func (w *tableWorker) executeTask(ctx context.Context, task *lookupTableTask) er
...
 		if len(w.idxLookup.tblPlans) == 1 {
+			obtainedHandlesMap := make(map[int64]struct{}, len(task.rows))
+			for _, row := range task.rows {
+				handle := row.GetInt64(w.handleIdx)
+				obtainedHandlesMap[handle] = struct{}{}
+			}
+
+			logutil.Logger(ctx).Error("inconsistent index handles", zap.String("index", w.idxLookup.index.Name.O),
+				zap.Int("index_cnt", handleCnt), zap.Int("table_cnt", len(task.rows)),
+				zap.Any("missing_handles", GetLackHandles(task.handles, obtainedHandlesMap)),
```
Comment (jackysp) [inline executor/distsql.go]: It's best to change to `Int64s`, but it doesn't matter.

Comment (qw4990) [reply executor/distsql.go]: fixed.

File: executor/distsql.go [hunk 4]
```diff
@@ -839,6 +839,17 @@ func (w *tableWorker) executeTask(ctx context.Context, task *lookupTableTask) er
...
+			obtainedHandlesMap := make(map[int64]struct{}, len(task.rows))
+			for _, row := range task.rows {
+				handle := row.GetInt64(w.handleIdx)
+				obtainedHandlesMap[handle] = struct{}{}
+			}
+
+			logutil.Logger(ctx).Error("inconsistent index handles", zap.String("index", w.idxLookup.index.Name.O),
+				zap.Int("index_cnt", handleCnt), zap.Int("table_cnt", len(task.rows)),
+				zap.Int64s("missing_handles", GetLackHandles(task.handles, obtainedHandlesMap)),
+				zap.Int64s("total_handles", task.handles))
```
Comment (lamxTyler) [inline executor/distsql.go]: Why log all the handles?

Comment (qw4990) [reply executor/distsql.go]: Used to help us construct handle ranges when investigating inconsistent problems.

**Comment by qw4990** [comment]
Wait for #10845.

**Comment by zz-jason** [comment]
@qw4990 Should we cherry pick this commit to release-3.0 and release-2.1?

**Pull request merged**